source/common/common/key_value_store_base.cc

```diff
@@ -58,20 +69,32 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
   return true;
 }
 
-void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_view value_view) {
+void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_view value_view,
+                                    absl::optional<std::chrono::milliseconds> ttl) {
   ENVOY_BUG(!under_iterate_, "addOrUpdate under the stack of iterate");
   std::string key(key_view);
   std::string value(value_view);
+  // Remove ttl if <= 0
+  if (ttl && ttl <= std::chrono::milliseconds(0)){
+    ASSERT(false);
+    ttl = absl::nullopt;
+  }
+
   // Attempt to insert the entry into the store. If it already exists, remove
   // the old entry and insert the new one so it will be in the proper place in
   // the linked list.
   if (!store_.emplace(key, value).second) {
     store_.erase(key);
     store_.emplace(key, value);
+    ttl_manager_.clear(key);
```

Comment:
Changed the behavior to clear any existing ttl if the key gets updated with ttl=null

Reply:
Do we have test coverage for updating the TTL from a valid TTL to null? Or from one ttl to another before the timer fires? If not, we should add it.